core/templates/dev/head/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.directive.ts

```diff
@@ -66,56 +67,59 @@ oppia.directive('adminDevModeActivitiesTab', [
           });
         };
 
-        ctrl.DEMO_EXPLORATIONS = GLOBALS.DEMO_EXPLORATIONS;
-        ctrl.DEMO_COLLECTIONS = GLOBALS.DEMO_COLLECTIONS;
         ctrl.numDummyExpsToPublish = 0;
         ctrl.numDummyExpsToGenerate = 0;
+        AdminDataService.getDataAsync().then(function(response) {
+          ctrl.DEMO_EXPLORATIONS = response.demo_explorations;
+          ctrl.DEMO_COLLECTIONS = response.demo_collections;
 
-        ctrl.reloadAllExplorations = function() {
-          if (AdminTaskManagerService.isTaskRunning()) {
-            return;
-          }
-          if (!$window.confirm('This action is irreversible. Are you sure?')) {
-            return;
-          }
-
-          ctrl.setStatusMessage('Processing...');
-          AdminTaskManagerService.startTask();
+          ctrl.reloadAllExplorations = function() {
```

Comment:
is there any specific reason to keep this method inside `then`?

Reply:
But this specific method doesn't use these values.

Reply:
This method is used in the UI, I don't think we should initialize a method on the async call. I think we should define it before and maybe allow calling this method only after the async call completes.

Reply:
Done